### PR TITLE
Reload orderscreen credit

### LINKED ISSUE #782: Cannot order on user after updating balance from negative to positive on second device
### How to reproduce

1. Make sure a user has a negative balance
1. Open the order screen on device A
3. Add a credit mutation to the user on device B such that the user's balance is positive again
4. Try to place an order for the user with negative (now positive) balance

### Expected behaviour

One can place the order for the user, since his balance is now positive

### Current behaviour

The old balance of the user is cached and one cannot place an order for the user until reloading the page

### PR DESCRIPTION
**Pull request by wilco375**
Fixes #782 
Reloads the user, and with that his balance and if ordering is allowed, when a user is selected in the orderscreen.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-class UsersController < ApplicationController
+class UsersController < ApplicationController # rubocop:disable Metrics/ClassLength
```
Comment (DrumsnChocolate) [inline app/controllers/users_controller.rb:1]: No way to shorten?

File: app/javascript/packs/order_screen.js [hunk 2]
```diff
@@ -67,6 +67,21 @@ document.addEventListener('turbolinks:load', () => {
             this.orderRows = [];
           }
 
+          if (user !== null) {
+            // Reload user to get latest credit balance
+            this.$http.get('/users/'+user.id).then((response) => {
+              const user = response.body;
+              this.$set(this.users, this.users.indexOf(user), user);
```
Comment (DrumsnChocolate) [inline app/javascript/packs/order_screen.js:74]: Afaik indexOf works with objects that are already present in the list, right? Dont you need to pass an additional function to indexOf to define when a match is found?

**Review by DrumsnChocolate — APPROVED** [review]

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (wilco375) [reply app/controllers/users_controller.rb:1]: Not in a way that makes it more readable I don't think

File: app/javascript/packs/order_screen.js (hunk 2 above)
Comment (wilco375) [reply app/javascript/packs/order_screen.js:74]: `indexOf` returns the index of the element (integer); `this.$set` then sets this index in the `this.users` array to `user`

**Pull request merged**